alphastats/loader/SpectronautLoader.py

```diff
@@ -38,74 +38,76 @@ def __init__(
         self.confidence_column = None
         self.filter_columns = []
         self.evidence_df = None
-        self.gene_names = gene_names_column
 
-        self._read_spectronaut_file(file=file, sep=sep)
-
-        is_long = self._check_if_long(self.rawinput)
+        columns = (
+            "^"
+            + "$|^".join(
+                [
+                    ".*" + intensity_column,
+                    index_column,
+                    sample_column,
+                    gene_names_column,
+                ]
+            )
+            + "$"
+        )
 
-        if filter_qvalue and is_long:
-            self._filter_qvalue(qvalue_cutoff=qvalue_cutoff)
+        self.rawinput = self._read_spectronaut_file(file=file, sep=sep, columns=columns)
 
-        if is_long:
-            self._reshape_spectronaut(
-                sample_column=sample_column, gene_names_column=gene_names_column
+        if gene_names_column in self.rawinput.columns.to_list():
+            self.gene_names = gene_names_column
+        else:
+            self.gene_names = None
+        if sample_column in self.rawinput.columns.to_list():
+            self.sample_column = sample_column
+        else:
+            self.sample_column = None
+            warnings.warn(
+                f"No sample column was found so the data will be treated as wide format. If this is wrong make sure {sample_column} is present in the file."
             )
 
+        self.rawinput = self._deduplicate_data(long=self.sample_column)
+
+        if self.sample_column is not None:
+            self.rawinput = self._reshape_long_to_wide()
+
+        self.intensity_column = "[sample]." + self.intensity_column
```

Comment:
Can do, but it's immutable because in wide format the data comes with the dot in the column name.

Reply:
alright, still I'd say this adds value (because it makes the relations between the different occurrences transparend [literally connects the dots :-D]) ..
mb call it "SPECTRONAUT_COLUMN_DELIM" then